### PR TITLE
[2.9] update for vsphere charts upgrade support

### PR DESCRIPTION
**Pull request by slickwarren**
## Issue: 

https://github.com/rancher/qa-tasks/issues/1291

## Problem

we need automated support for upgrading vsphere charts

### CODE REVIEW

File: tests/v2/validation/provisioning/rke1/README.md [hunk 1]
```diff
@@ -67,6 +67,8 @@ provisioningInput:
         secretKey: ""
     retention: "72h"
     snapshot: false
+chartUpgrade:
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke1/README.md:70]: nit; can you leave a comment explaining this a bit more? I don't think this is immediately intuitive to someone new to this test.

File: tests/v2/validation/provisioning/rke1/provisioning_cloud_provider_test.go [hunk 2]
```diff
@@ -104,7 +106,7 @@ func (r *RKE1CloudProviderTestSuite) TestAWSCloudProviderRKE1Cluster() {
 
 func (r *RKE1CloudProviderTestSuite) TestVsphereCloudProviderRKE1Cluster() {
 	nodeRolesDedicated := []provisioninginput.NodePools{provisioninginput.EtcdNodePool, provisioninginput.ControlPlaneNodePool, provisioninginput.WorkerNodePool}
-	nodeRolesDedicated[0].NodeRoles.Quantity = 3
+	nodeRolesDedicated[0].NodeRoles.Quantity = 1
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke1/provisioning_cloud_provider_test.go:109]: Why is this changed to 1?

File: tests/v2/validation/provisioning/rke2/README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ provisioningInput:
   rke2KubernetesVersion: ["v1.27.10+rke2r1"]
   cni: ["calico"]
   providers: ["linode", "aws", "do", "harvester", "vsphere", "azure"]
-  cloudProvider: "aws"
+  cloudProvider: "aws" # rancher-vsphere
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/README.md:58]: nit - can you update the comment to say something like `For vSphere, enter rancher-vsphere`

File: tests/v2/validation/upgrade/README.md [hunk 4]
```diff
@@ -53,4 +53,38 @@ rancher:
...
+```yaml
+rancher:
+  host: <your_host>
+  adminToken: <your_token>
+  insecure: true/false
+  cleanup: false/true
+  clusterName: "<your_cluster_name>"
+
+
+vmwarevsphereCredentials:
```
Comment (markusewalker) [inline tests/v2/validation/upgrade/README.md:84]: nit - can you have these bocks not as spaced out? Not a deal breaker, just used to seeing YAML not have as much (if any) white space in our tests.

**Review by markusewalker — COMMENTED** [review]
Overall, looks fine. Left some nits/comments.

File: tests/v2/validation/provisioning/rke1/README.md (hunk 1 above)
Comment (igomez06) [reply tests/v2/validation/provisioning/rke1/README.md:70]: I agree I still don't understand why we need this new configuration.

File: tests/v2/validation/provisioning/rke1/provisioning_cloud_provider_test.go (hunk 2 above)
Comment (slickwarren) [reply tests/v2/validation/provisioning/rke1/provisioning_cloud_provider_test.go:109]: I was running into trouble with resource usage in vsphere 7 setup, and technically you don't need 3 etcd for this test. Good callout though I can change it back to 3.

File: tests/v2/validation/provisioning/rke1/README.md (hunk 1 above)
Comment (slickwarren) [reply tests/v2/validation/provisioning/rke1/README.md:70]: added a comment here, lmk if it is more clear. But for more context, this is the main feature of this PR - allowing the chart to be upgraded via automation. Before, we just installed the latest version of the chart by default in automation. If that's the case, we can't upgrade it later. So this option installs latest - 1 so that the chart can be upgraded in a later test.

File: tests/v2/validation/provisioning/rke1/README.md [hunk 5]
```diff
@@ -45,10 +45,10 @@ provisioningInput:
       worker: true
       drainBeforeDelete: true
       quantity: 2
-  rke1KubernetesVersion: ["v1.27.10-rancher1-1"]
+  rke1KubernetesVersion: ["v1.28.10-rancher1-1"]
   cni: ["calico"]
   providers: ["linode", "aws", "do", "harvester", "vsphere", "azure"]
-  cloudProvider: "external-aws"
+  cloudProvider: "" # either: external-aws|rancher-vsphere
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke1/README.md:51]: Can this be expanded upon more? Something like `Currently provides support for AWS and vSphere. The supported values are `external-aws | rancher-vsphere`.

File: tests/v2/validation/provisioning/rke2/README.md [hunk 6]
```diff
@@ -55,7 +55,7 @@ provisioningInput:
   rke2KubernetesVersion: ["v1.27.10+rke2r1"]
   cni: ["calico"]
   providers: ["linode", "aws", "do", "harvester", "vsphere", "azure"]
-  cloudProvider: "aws"
+  cloudProvider: "" # either: aws|rancher-vsphere
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/README.md:58]: Same as above.

File: tests/v2/validation/provisioning/rke1/README.md (hunk 5 above)
Comment (slickwarren) [reply tests/v2/validation/provisioning/rke1/README.md:51]: we don't usually provide that much context in the comments of the readme. i.e. `providers` and `psact` don't have reasoning for what's listed

Comment (slickwarren) [reply tests/v2/validation/provisioning/rke1/README.md:51]: also there's a whole section later on for what is supported, so I don't feel that adding further comments here is helpful. wdyt?

Comment (markusewalker) [reply tests/v2/validation/provisioning/rke1/README.md:51]: I don't feel strongly of it. I was more looking at perspective if someone doesn't read the other section. If you think it's fine to leave as is, that's fine with me.

**Review by Josh-Diamond — APPROVED** [review]

**Review by susesgartner — APPROVED** [review]
LGTM
